d2/src/main/java/com/linkedin/d2/balancer/dualread/DualReadLoadBalancer.java

```diff
@@ -156,6 +156,12 @@ public void getClient(Request request, RequestContext requestContext, Callback<T
         _newLb.getClient(request, requestContext, clientCallback);
         break;
       case DUAL_READ:
+        if (_newLbExecutor.isShutdown())
+        {
+          _rateLimitedLogger.error("Dual read failure. New load balancer executor is shutdown.");
+          _oldLb.getClient(request, requestContext, clientCallback);
+          return;
+        }
         _newLbExecutor.execute(
```

Comment:
all usages of _newLbExecutor in this file need the same check.

Reply:
Thanks for reminder. Updated